tests/director/tests/bundle_instance_auth_per_scenario_test.go

```diff
@@ -0,0 +1,290 @@
+package tests
+
+import (
+	"context"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
+	"github.com/kyma-incubator/compass/tests/pkg/fixtures"
+	"github.com/kyma-incubator/compass/tests/pkg/gql"
+	"github.com/kyma-incubator/compass/tests/pkg/ptr"
+	"github.com/kyma-incubator/compass/tests/pkg/tenant"
+	"github.com/kyma-incubator/compass/tests/pkg/testctx"
+	"github.com/kyma-incubator/compass/tests/pkg/token"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUnassignResourceFromScenarioWhenThereIsExistingBundleInstanceAuth(t *testing.T) {
+	ctx := context.Background()
+	createScenarioLabelDefIfNotExist(t, ctx)
+
+	commonScenario := "common-scenario"
+	otherAppScenario := "other-app-scenario"
+	otherRuntimeScenario := "other-runtime-scenario"
+
+	scenarios := []string{conf.DefaultScenario, commonScenario, otherAppScenario, otherRuntimeScenario}
+	fixtures.UpdateScenariosLabelDefinitionWithinTenant(t, ctx, dexGraphQLClient, tenant.TestTenants.GetDefaultTenantID(), scenarios)
+	defer fixtures.UpdateScenariosLabelDefinitionWithinTenant(t, ctx, dexGraphQLClient, tenant.TestTenants.GetDefaultTenantID(), scenarios[:1])
+
+	// GIVEN
+	data, cleanup, ok := setupAppAndRuntimeWithCommonScenario(t, ctx, commonScenario)
+	defer cleanup()
+	require.True(t, ok)
+
+	t.Run("Remove Application from scenario by setApplicationLabel mutation", func(t *testing.T) {
+		request := fixtures.FixSetApplicationLabelRequest(data.appId, ScenariosLabel, []interface{}{otherAppScenario})
+		label := graphql.Label{}
+		err := testctx.Tc.RunOperation(ctx, dexGraphQLClient, request, &label)
+		require.Error(t, err)
```

Comment:
Valid for all the tests.

I think that we should require a specific error here since those are e2e tests and not unit tests plenty of errors can occur (networking, credentials, communication, etc...). This way the test can be false positive which is risky for e2e tests.